homeassistant/components/light/xiaomi_miio.py

```diff
@@ -36,25 +46,38 @@
 CCT_MIN = 1
 CCT_MAX = 100
 
+DELAYED_TURN_OFF_MAX_DEVIATION = 4
+
 SUCCESS = ['ok']
 ATTR_MODEL = 'model'
 ATTR_SCENE = 'scene'
+ATTR_DELAYED_TURN_OFF = 'delayed_turn_off'
+ATTR_SECONDS = 'seconds'
 
 SERVICE_SET_SCENE = 'xiaomi_miio_set_scene'
+SERVICE_SET_DELAYED_TURN_OFF = 'xiaomi_miio_set_delayed_turn_off'
 
 XIAOMI_MIIO_SERVICE_SCHEMA = vol.Schema({
     vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
 })
 
-SERVICE_SCHEMA_SCENE = XIAOMI_MIIO_SERVICE_SCHEMA.extend({
+SERVICE_SCHEMA_SET_SCENE = XIAOMI_MIIO_SERVICE_SCHEMA.extend({
     vol.Required(ATTR_SCENE):
         vol.All(vol.Coerce(int), vol.Clamp(min=1, max=4))
 })
 
+SERVICE_SCHEMA_SET_DELAYED_TURN_OFF = XIAOMI_MIIO_SERVICE_SCHEMA.extend({
+    vol.Required(ATTR_SECONDS):
+        vol.All(vol.Coerce(int), vol.Range(min=0))
```

Comment:
Use `vol.All(cv.time_period, cv.positive_timedelta)`. That allows the user to specify in multiple formats.

You can then in your code just call `call.data[ATTR_SECONDS].total_seconds()`

Reply:
Cool!